### PR TITLE
20286 - fix protocol tests

### PR DESCRIPTION
**Pull request by NissimHadar**
https://highfidelity.fogbugz.com/f/cases/20286/

# Description
1.  The PolyVox protocol test is wrong.  The test was failing, but nitpick didn't catch the error due to a bug in the test validation code.

2.  The result of the `render / engine / camera / primary` test depends on the size of the avatar.  Modify to use fixed avatar
# Testing
1.  Stop any running instances  of Sandbox and Interface.
2.  Create a folder for test results (e.g. - D:\tests)
3.  On Windows - open a terminal in the High Fidelity program folder.
4.  Run the sandbox 
5.  Run the following commands:
`start domain-server`
`start assignment-client -n 6`
`interface.exe --url hifi://localhost --no-updater --no-login-suggestion --testScript https://raw.githubusercontent.com/NissimHadar/hifi_tests/20286-fixProtocolTests/tests/protocol/polyvox/testAuto.js quitWhenFinished --testResultsLocation D:/temp`.  
Replace D:/temp with the test results folder created in step 1
6.  After completion, verify that a file named `tests.protocol.polyvox.00000.passed.txt` exists in the test results folder
7.  Delete `tests.protocol.polyvox.00000.passed.txt`
8.  Run the following command:
`interface.exe --url hifi://localhost --no-updater --no-login-suggestion --testScript https://raw.githubusercontent.com/NissimHadar/hifi_tests/20286-fixProtocolTests/tests/engine/render/camera/primary/testAuto.js quitWhenFinished --testResultsLocation D:/temp`.  
9.  Use nitpick v1.2 to evaluate the results.

### CODE REVIEW

**Comment by qaAdamSmith82** [comment]
Currently Testing

**Comment by qaAdamSmith82** [comment]
When I run "start assignment-client -n 6" I continually get this message:
[12/12 16:06:34] [DEBUG] [hifi.networking] [4840] [assignment-client] udt::Socket error -  QAbstractSocket::ConnectionRefusedError
[12/12 16:06:34] [DEBUG] [hifi.networking] [10864] [assignment-client] udt::Socket error -  QAbstractSocket::ConnectionRefusedError

If I skip that step and just run:

interface.exe --url hifi://localhost --no-updater --no-login-suggestion --testScript https://raw.githubusercontent.com/NissimHadar/hifi_tests/20286-fixProtocolTests/tests/protocol/polyvox/testAuto.js quitWhenFinished --testResultsLocation C:/temp.

Then the protocol test fails, see attached files
[tests.protocol.polyvox.00000.failed.txt](https://github.com/highfidelity/hifi_tests/files/2674016/tests.protocol.polyvox.00000.failed.txt)

Is there an addition setting I need to flag to get past the socket error?

**Comment by NissimHadar** [comment]
This could happen if Sandbox is running.  I updated the procedure to stop any instances of Sandbox/Interface that are running.

The test _will_ fail without a server, that is by design.

File: tests/protocol/polyvox/test.js [hunk 1]
```diff
@@ -14,7 +14,6 @@ nitpick.perform("Polyvox protocol sanity - TEST REQUIRES SERVER", Script.resolve
     entityProperties.dimensions = { x: 2000.0, y: 2000.0, z: 2000.0 };
     entityProperties.voxelVolumeSize = { x:23, y: 29, z: 31 };
     entityProperties.voxelData="ads23SGResLiIUYoh987fTREdgdgDGdgdgDSDLLIJklj65edrcgddQ7Y"
-    entityProperties.PolyVoxSurfaceStyle = 3;
```
Comment (SamGondelman) [inline tests/protocol/polyvox/test.js]: We aren't testing voxelSurfaceStyle now.

File: tests/protocol/common.js [hunk 2]
```diff
@@ -81,13 +79,14 @@ compareObjects = function(object1, object2, failures, elementName) {
 }
 
 saveResults = function(failures) {
-    if (allPassed(failures)) {
+	allPassedFlag = allPassed(failures);
```
Comment (SamGondelman) [inline tests/protocol/common.js]: tab

File: tests/protocol/polyvox/test.js (hunk 1 above)
Comment (NissimHadar) [reply tests/protocol/polyvox/test.js]: I know.... When I set this, I get a null (undefined) value on reading the value back.  Do you have an idea why?

Comment (SamGondelman) [reply tests/protocol/polyvox/test.js]: "voxelSurfaceStyle" not "PolyVoxSurfaceStyle"

Comment (NissimHadar) [reply tests/protocol/polyvox/test.js]: Ah ... the code comments are very confusing:
Line 1066 in EntityItemProperties:
` * @property {Entities.PolyVoxSurfaceStyle} voxelSurfaceStyle=2 - The style of rendering the voxels' surface and how `
Line 82 in PolyVoxEntityItem:
`     * @typedef {number} Entities.PolyVoxSurfaceStyle`

Now I see that is the type, not the name......

Fixed.

**Pull request merged**